src/plugins/intel_cpu/src/nodes/concat.cpp

```diff
@@ -342,6 +342,41 @@ void Concat::prepareParams() {
             hasOuterLoop = true;
         }
     }
+
+    execSpecialCase = nullptr;
+    if (outputShape.size() == 1 && outputStrides[0] == 1 && outputShape[0] <= 64 && elemSize == 4) {
+        // output is small 1d vector (which is typical in shape inference subgraph),
+        // in this case, inputs are also small 1d vector and single thread naive impl is faster
+        bool useSpecial = true;
+        for (size_t i = 0; i < getParentEdges().size(); i++) {
+            const auto& srcMemPtr = getParentEdgesAtPort(i)[0]->getMemoryPtr();
+            const auto srcMemDesc = srcMemPtr->getDescPtr()->as<BlockedMemoryDesc>();
+            const auto& inputShape = srcMemDesc->getBlockDims();
+            const auto& strides = srcMemDesc->getStrides();
+            if (inputShape.size() != 1 || strides.size() != 1) {
+                useSpecial = false;
+                break;
+            }
+        }
+        if (useSpecial) {
+            execSpecialCase = [this]() {
```

Comment:
Can we extract it into separate function? Just to have aligned implementation with other special impls.
Same comment for other files.